feat: Pebble Notices (#20)
canonical/gocert: README.md

* A private key
* A TLS certificate with that private key
* A YAML config file with the required parameters

> you can generate the cert and the associated key by running:

Reviewer: ```suggestion
You can generate the cert and the associated key by running:
```

`openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 1 -out cert.pem -subj "/CN=example.com"`
GoCert does not support insecure http connections.

### Config File
The config file requires the following parameters:

> * keypath: this is the path to the private key you've generated.

Reviewer 2: ```suggestion
* keypath: path to the private key for enabling HTTPS connections (string).
```

Author: Took all of the suggestions and converted this into a table. [Here's](https://github.com/canonical/gocert/blob/TLSENG-239-pebble-notices/README.md) the preview.

> * certpath: this is the path to a certificate for enabling HTTPS connections.

Reviewer 2: ```suggestion
* certpath: path to a PEM formatted certificate for enabling HTTPS connections (string).
```

Author: Took all of the suggestions and converted this into a table. [Here's](https://github.com/canonical/gocert/blob/TLSENG-239-pebble-notices/README.md) the preview.

> * dbpath: the path to a sqlite database file. If the file does not exist GoCert will attempt to create it.

Reviewer 2: ```suggestion
* dbpath: path to a sqlite database file. If the file does not exist GoCert will attempt to create it. (string)
```

Author: Took all of the suggestions and converted this into a table. [Here's](https://github.com/canonical/gocert/blob/TLSENG-239-pebble-notices/README.md) the preview.

> * port: the port in integer form to serve all of GoCert's API and frontend

Reviewer 2: ```suggestion
* port: port number on which GoCert will listen for all incoming API and frontend connections (integer).
```

> * pebblenotificationsenabled: a boolean that once enabled, will allow GoCert to send pebble notices. Read more about it (here)[https://github.com/canonical/pebble?tab=readme-ov-file#notices].

Reviewer 2: Adding underscores would improve readability. I'd recommend `pebble_notify` instead of `pebblenotificationsenabled`. The same applies to other parameters (ex. `key_path` instead of `keypath`), but those are likely not part of this PR's scope

Author: changed all the params here, since it's pretty easy to remap them. The config stuff can use some cleaning up still, I've opened another ticket for that. TLSENG-270

> * pebblenotificationsenabled: a boolean that once enabled, will allow GoCert to send pebble notices. Read more about it (here)[https://github.com/canonical/pebble?tab=readme-ov-file#notices].

Reviewer 2: ```suggestion
* pebble_notify: Allow GoCert to send pebble notices on certificate events (create, update, delete). Read more about it (here)[https://github.com/canonical/pebble?tab=readme-ov-file#notices]. (bool)
```

Author: Took all of the suggestions and converted this into a table. [Here's](https://github.com/canonical/gocert/blob/TLSENG-239-pebble-notices/README.md) the preview.

> an example config file may look like:

Reviewer: ```suggestion
An example config file may look like:
```

* A private key
* A TLS certificate with that private key
* A YAML config file with the required parameters

You can generate the cert and the associated key by running:
```
openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 1 -out cert.pem -subj "/CN=example.com"
```

GoCert does not support insecure http connections.

### Config File
The config file requires the following parameters:
| Key                  | Type    | Description |
|----------------------|---------|----------|
| key_path             | string  | path to the private key for enabling HTTPS connections |
| cert_path            | string  | path to a PEM formatted certificate for enabling HTTPS connections |
| db_path              | string  | path to a sqlite database file. If the file does not exist GoCert will attempt to create it. |

> | port                 | number  | port number on which GoCert will listen for all incoming API and frontend connections. |

Reviewer 2: ```suggestion
| port                 | integer  | port number on which GoCert will listen for all incoming API and frontend connections. |
```

> | pebble_notifications | boolean | Allow GoCert to send pebble notices on certificate events (create, update, delete). Read more about it [here](https://github.com/canonical/pebble?tab=readme-ov-file#notices). |

Reviewer 2: ```suggestion
| pebble_notifications | boolean | Allow GoCert to send pebble notices on certificate events (create, update, delete). Pebble needs to be running on the same system as GoCert. Read more about Pebble Notices [here](https://github.com/canonical/pebble?tab=readme-ov-file#notices). |
```
